packages/signature-controller/src/SignatureController.ts

```diff
@@ -250,6 +251,27 @@ export class SignatureController extends BaseControllerV2<
     return this.#typedMessageManager.getUnapprovedMessagesCount();
   }
 
+  /**
+   * Retrieves an array of all messages from TypedMessages and PersonalMessages.
+   *
+   * @returns The array of all messages from their proxies.
+   */
+  get allMessages(): unknown[] {
+    let messagesObject: any = {};
+
+    const allMessages = [
+      ...this.#typedMessageManager.getAllMessages(),
+      ...this.#personalMessageManager.getAllMessages(),
+      ...this.#messageManager.getAllMessages(),
+    ];
+
+    for (const message of allMessages) {
```

Comment:
Is there any benefit to having the array and then iterating over it, could we not just directly return a new object and use the spread operator on the three `getAllMessages` methods directly?